client/components/UpdateZoopDialog.tsx

```diff
@@ -0,0 +1,101 @@
+import SendIcon from "@mui/icons-material/Send"
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import TextField from "@mui/material/TextField";
+import React, { useState } from "react";
+import { useSelector } from "react-redux";
+
+import { useUpdateZoopMutation } from "../features/api";
+import { RootState } from "../app/store";
+import { useNavigate } from "react-router-dom";
+import { ZoopWithDetails } from "../../src/types/custom";
+import UpdateSnackbar from "./UpdateSnackbar";
+
+export interface UpdateZoopDialogProps {
+    open: boolean
+    onClose: () => void
+    zoop: ZoopWithDetails
+}
+
+const UpdateZoopDialog = ({open, onClose, zoop}: UpdateZoopDialogProps) => {
+    const [contentForUpdate, setContentForUpdate] = useState(zoop.content);
+    const [isSuccessSnackbarOpen, setIsSuccessSnackbarOpen] = useState(false);
+
+    const navigate = useNavigate();
+
+    const currentUser = useSelector((state: RootState) => state.auth.user)
+    
+    const [updateZoop, { isLoading, isError, data, error }] = useUpdateZoopMutation();
+
+    if (!currentUser) {
+        console.error("User not logged in does not have access to update zoop dialog");
+        navigate('/login');
+        return null;
+    }
+
+
+    const handleUpdateZoopClick = async () => {
+        const updatedZoop = await updateZoop({
```

Comment:
```suggestion
    const handleUpdateZoopClick = async (event) => {
        event.stopPropagation();
        const updatedZoop = await updateZoop({
```

Reply:
It actually worked without me having to do this.